apps/meteor/ee/app/livechat-enterprise/server/priorities.ts

```diff
@@ -0,0 +1,39 @@
+import { LivechatPriority } from '@rocket.chat/models';
+import type { ILivechatPriority } from '@rocket.chat/core-typings';
+import { LivechatPriorityWeight } from '@rocket.chat/core-typings';
+
+const defaultPriorities: Omit<ILivechatPriority, '_id' | '_updatedAt'>[] = [
+	{
+		i18n: 'Lowest',
+		sortItem: LivechatPriorityWeight.LOWEST,
+		dirty: false,
+	},
+	{
+		i18n: 'Low',
+		sortItem: LivechatPriorityWeight.LOW,
+		dirty: false,
+	},
+	{
+		i18n: 'Medium',
+		sortItem: LivechatPriorityWeight.MEDIUM,
+		dirty: false,
+	},
+	{
+		i18n: 'High',
+		sortItem: LivechatPriorityWeight.HIGH,
+		dirty: false,
+	},
+	{
+		i18n: 'Highest',
+		sortItem: LivechatPriorityWeight.HIGHEST,
+		dirty: false,
+	},
+];
+
+export const createDefaultPriorities = async (): Promise<void> => {
+	const priorities = await LivechatPriority.find().toArray();
```

Comment:
there is no need to fetch ALL data just to check if a single record exists.. imagine how much data will be transferred and parsed just to be thrown away.. 

you have two options here:

- you can use `.countDocuments()` to count and check if the result is equal to zero
- you can query using `findOne()` without a filter using `projection: { _id: 1 }` and check if the result is not `null`.. this is the preferred method

Reply:
In normal cases I'd say yes. But for this case, LivechatPriority is expected to have 5 and only 5 elements, so we went this way.

We can change to a countDocuments too, but yeah, not so concerning here given the size of the collection